extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -116,133 +107,126 @@ public function renderInfobox( array $infoboxdata, $theme, $layout ) {
+			$data[$i] = $helper->extendImageData( $data[$i], $this->imagesWidth );
+
+			if ( !!$data[$i] ) {
+				$images[] = $data[$i];
 			}
-		} elseif ( !$helper->isMercury() || $firstInfoboxAlredyRendered ) {
-			return $this->renderItem( 'title', $data[ 'title' ] );
 		}
 
-		return '';
+		if ( count( $images ) === 0 ) {
+			return '';
+		}
+
+		if ( count( $images ) === 1 ) {
+			$data = $images[0];
+			$templateName = 'image';
+		} else {
+			// More than one image means image collection
+			$data = $helper->extendImageCollectionData( $images );
+			$templateName = 'image-collection';
+		}
+
+		return $this->render( $templateName, $data );
 	}
 
-	/**
-	 * renders part of infobox
-	 * If image element has invalid thumbnail, doesn't render this element at all.
-	 *
-	 * @param string $type
-	 * @param array $data
-	 *
-	 * @return bool|string - HTML
-	 */
-	private function renderItem( $type, array $data ) {
-		$helper = new PortableInfoboxRenderServiceHelper();
+	protected function renderTitle( $data ) {
+		$data['inlineStyles'] = $this->inlineStyles;
 
-		if ( $type === 'image' ) {
-			$images = [ ];
+		return $this->render( 'title', $data );
+	}
 
-			for ( $i = 0; $i < count( $data ); $i++ ) {
-				$data[ $i ][ 'context' ] = self::MEDIA_CONTEXT_INFOBOX;
-				$data[ $i ] = $helper->extendImageData( $data[ $i ], $this->imagesWidth );
+	protected function renderHeader( $data ) {
+		$data['inlineStyles'] = $this->inlineStyles;
 
-				if ( !!$data[ $i ] ) {
-					$images[] = $data[ $i ];
-				}
+		return $this->render( 'header', $data );
+	}
+
+	protected function renderChildren( $children ) {
+		$result = '';
+		foreach ( $children as $child ) {
+			$type = $child['type'];
+			if ( $this->templateEngine->isSupportedType( $type ) ) {
+				$result .= $this->renderItem( $type, $child['data'] );
 			}
+		}
 
-			if ( count( $images ) === 0 ) {
-				return false;
-			} else if ( count( $images ) === 1 ) {
-				$data = $images[ 0 ];
-				$templateName = $type;
-			} else {
-				// More than one image means image collection
-				if ( $helper->isMobile() && !$helper->isMercury() ) {
-					// Display only the first image on WikiaMobile
-					$data = $images[ 0 ];
-				} else {
-					$data = $helper->extendImageCollectionData( $images );
-				}
+		return $result;
+	}
 
-				$templateName = 'image-collection';
-			}
+	private function getInlineStyles( $accentColor, $accentColorText ) {
+		$backgroundColor = empty( $accentColor ) ? '' : "background-color:{$accentColor};";
+		$color = empty( $accentColorText ) ? '' : "color:{$accentColorText};";
+
+		return "{$backgroundColor}{$color}";
+	}
+
+	private function createHorizontalGroupData( $groupData ) {
+		$horizontalGroupData = [
+			'labels' => [ ],
+			'values' => [ ],
+			'renderLabels' => false
+		];
+
+		foreach ( $groupData as $item ) {
+			$data = $item['data'];
+
+			if ( $item['type'] === 'data' ) {
+				array_push( $horizontalGroupData['labels'], $data['label'] );
```

Comment:
Is it possible to have 2nd argument type other than string?
If no can we simply use `$horizontalGroupData['labels'][] = $data['label'];`?